src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+import java.time.LocalDate;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.ListCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+public class Parser {
+
+    /**
+     * Takes in user's input and convert it into a command obj.
+     *
+     * @param input User's raw input
+     * @return A command object
+     * @throws IncompleteCommandException if command is incomplete
+     */
+    public static Command parse(String input) throws IncompleteCommandException {
+        if (input.toUpperCase().equals(CommandList.BYE.toString())) {
```

Comment:
Should the input.toUpperCase() in lines 28, 30, 32, 36, 55 and 73 be put into a separate line to allow for better readability?